model_2/db_supabase_connect.js

```diff
@@ -0,0 +1,40 @@
+const {createClient} =  "@supabase/supabase-js"
+const { Pool } = require('pg');
+
+// const supabaseUrl = env.SB_URL
+// const supabaseKey = 
+// const supabase = createClient(supabaseUrl, supabaseKey)
+// console.log(supabase)
+// module.exports = {supabase}
+
+
+let client = null
+async function getClient () {
+  if (client) {
+    return client
+  }
+  try {
+    client = new Pool({
+      supabaseUrl: env.PG_URL,
+      supabaseKey: process.env.PG_KEY,
+      supabase:  createClient(supabaseUrl, supabaseKey)
+
+        });
+    await client.connect()
+    return client
+  } catch (err) {
+    console.log('Error when connecting to PG:')
+    console.log(err.message)
+    return null
+  }
+}
+
+async function query (text) {
+    const client = await getClient()
+    return client.query(text)
+  }
+
+
+getClient()
```

Comment:
You don't need to call this function at the bottom. It should only pass the function to be used elsewhere.